Add a note to the Kafka tutorial about restarting for the `apply_function` (#3382)
astronomer/docs: learn/airflow-kafka.md

    The AwaitMessageTriggerFunctionSensor always runs and listens. If the task fails, like if a malformed message is consumed, the DAG completes as `failed` and automatically starts its next DAG run because of the [`@continuous` schedule](scheduling-in-airflow.md#continuous-timetable).

:::info

> Since the function provided to the `apply_function` parameter is being imported into the [Triggerer](https://docs.astronomer.io/learn/deferrable-operators#terms-and-concepts) compoment of Airflow, which does not periodically restart, you need to restart your Airflow instance to apply changes to the `apply_function` of the AwaitMessageTriggerFunctionSensor when working locally. To do so run `astro dev restart` in your terminal. Changes to the `event_triggered_function` of the AwaitMessageTriggerFunctionSensor do not require a restart of your Airflow instance.

Reviewer: ```suggestion
When working locally, you need to restart your Airflow instance to apply changes to the `apply_function` of the AwaitMessageTriggerFunctionSensor because the function is imported into the [Triggerer](https://docs.astronomer.io/learn/deferrable-operators#terms-and-concepts), which does not periodically restart. To do so, run `astro dev restart` in your terminal. Changes to the `event_triggered_function` of the AwaitMessageTriggerFunctionSensor do not require a restart of your Airflow instance.
```

    The AwaitMessageTriggerFunctionSensor always runs and listens. If the task fails, like if a malformed message is consumed, the DAG completes as `failed` and automatically starts its next DAG run because of the [`@continuous` schedule](scheduling-in-airflow.md#continuous-timetable).

:::info

> When working locally, you need to restart your Airflow instance to apply changes to the `apply_function` of the AwaitMessageTriggerFunctionSensor because the function is imported into the [Triggerer](https://docs.astronomer.io/learn/deferrable-operators#terms-and-concepts), which does not periodically restart. To do so, run `astro dev restart` in your terminal. Changes to the `event_triggered_function` of the AwaitMessageTriggerFunctionSensor do not require a restart of your Airflow instance.

Reviewer 2: ```suggestion
When working locally, you need to restart your Airflow instance to apply changes to the `apply_function` of the AwaitMessageTriggerFunctionSensor because the function is imported into the [Triggerer](https://docs.astronomer.io/learn/deferrable-operators#terms-and-concepts), which does not periodically restart. To restart Airflow, run `astro dev restart` in your terminal. Changes to the `event_triggered_function` of the AwaitMessageTriggerFunctionSensor do not require a restart of your Airflow instance.
```
